src/utils/unit_test/random_test.cpp

```diff
@@ -0,0 +1,46 @@
+// MUST include this
+#include <catch2/catch.hpp>
+
+// File being tested
+#include <lbann/utils/random.hpp>
+
+constexpr size_t num_tests = 1000;
+
+TEST_CASE("Testing fast_random_uniform", "[random][utilities]") {
+  SECTION("32-bit generator") {
+    std::mt19937 gen;
+    SECTION("floats") {
+      for (size_t i = 0; i < num_tests; ++i) {
+        float val = lbann::fast_random_uniform<float>(gen);
+        REQUIRE(val >= 0.0f);
+        REQUIRE(val <= 1.0f);
```

Comment:
```suggestion
        REQUIRE(val < 1.0f);
```
The other tests should be similarly changed.